api/node/package.json

```diff
@@ -16,15 +16,18 @@
     "@types/node-fetch": "^2.6.7",
     "ava": "^5.3.0",
     "jimp": "^0.22.8",
-    "typedoc": "^0.25.2"
+    "typedoc": "^0.25.2",
+    "esbuild": "^0.19.5",
+    "typescript": "^5.2.2"
   },
   "engines": {
     "node": ">= 10"
   },
   "scripts": {
     "artifacts": "napi artifacts",
-    "compile": "esbuild index.ts --bundle --external:*.node --format=cjs --platform=node --outfile=index.js && tsc --declaration --emitDeclarationOnly",
-    "build": "napi build --platform --release --js rust-module.js --dts rust-module.d.ts && npm run compile",
+    "compile_dts": "tsc index.ts --target esnext --module nodenext --moduleResolution nodenext --declaration --emitDeclarationOnly",
+    "compile": "esbuild index.ts --bundle --external:*.node --format=cjs --platform=node --outfile=index.js && npm run compile_dts",
```

Comment:
The fact that tsconfig configures the compiler differently than when calling it to emit the declarations is asking for trouble. Then we add esbuild to the mix and we have three different ways of "building" the .ts files.

This is not a regression of this change, but we should really try to streamline our tooling before the next release. I'm thinking ideally we could use just tsc.

Reply:
I agree. We should use tsc only.